inc/import/parser/class.ParserFITSingle.php

```diff
@@ -10,668 +10,134 @@
-        'saturated_hemoglobin_percent' => ['smo2_0', 0.1],
-        'total_hemoglobin_conc' => ['thb_0', 1]
-	);
-
-	/** @var array */
-	protected $nativeFieldMappingForRecord = array(
-		// TODO: native fields aller verwendeten Arrays
-		3 => ['heart_rate'],
-		4 => ['cadence', ['default' => 1, 'SPM' => 0.5]],
-		5 => ['distance', ['default' => 1, 'm' => 100]],
-		7 => ['power'],
-		39 => ['vertical_oscillation', ['default' => 1, 'Centimeters' => 100]],
-		54 => [['thb_0', 'thb_1'], 100],
-		57 => [['smo2_0', 'smo2_1'], 1]
-	);
-
-	/** @var array */
-	protected $DeveloperFieldMappingForSession = array();
-
-	/** @var array */
-	protected $nativeFieldMappingForSession = array(
-		9 => ['total_distance', ['default' => 1, 'm' => 100]],
-		11 => ['total_calories'],
-		44 => ['pool_length', ['default' => 1, 'm' => 100]],
-	);
-
-	/** @var array */
-	protected $DeveloperFieldMappingForLap = array();
-
-	/** @var array */
-	protected $nativeFieldMappingForLap = array(
-		9 => ['total_distance', ['default' => 1, 'm' => 100]],
-	);
-
-	protected function readFieldDescription() {
-		switch ($this->Values['native_mesg_num'][1]) {
-			case 'record':
-				$this->readFieldDescriptionFor($this->nativeFieldMappingForRecord, $this->DeveloperFieldMappingForRecord);
-				break;
-			case 'session':
-				$this->readFieldDescriptionFor($this->nativeFieldMappingForSession, $this->DeveloperFieldMappingForSession);
-				break;
-			case 'lap':
-				$this->readFieldDescriptionFor($this->nativeFieldMappingForLap, $this->DeveloperFieldMappingForLap);
-				break;
-		}
-	}
-
-	protected function readFieldDescriptionFor(array &$nativeFieldMapping, array &$fieldMapping) {
-		if (
-			isset($this->Values['native_field_num']) &&
-			isset($nativeFieldMapping[$this->Values['native_field_num'][0]]) &&
-			!empty($nativeFieldMapping[$this->Values['native_field_num'][0]][0]) &&
-			isset($this->Values['developer_data_index']) &&
-			isset($this->Values['field_name'])
-		) {
-			$fieldname = $this->Values['developer_data_index'][0].'_'.str_replace(['"', ' '], ['', '_'], $this->Values['field_name'][0]);
-			$fieldname = preg_replace_callback('/(\W)/i', function(array $char) {
-			    return sprintf('_%02x_', ord($char[0]));
-			}, preg_replace('/(\s+)/i', '_', $fieldname));
-			
-			$nativeFieldNum = $this->Values['native_field_num'][0];
-			$unitDefinition = str_replace('"', '', $this->Values['units'][0]);
-
-			$mappingKey = $nativeFieldMapping[$nativeFieldNum][0];
-			$mappingFactor = isset($nativeFieldMapping[$nativeFieldNum][1]) ? $nativeFieldMapping[$nativeFieldNum][1] : 1;
-
-			if (is_array($mappingFactor)) {
-				$mappingFactor = isset($mappingFactor[$unitDefinition]) ? $mappingFactor[$unitDefinition] : $mappingFactor['default'];
-			}
-
-			if (is_array($mappingKey)) {
-				$mappingKey = array_shift($nativeFieldMapping[$nativeFieldNum][0]);
-			}
-
-			$fieldMapping[$fieldname] = [$mappingKey, $mappingFactor];
-		}
+	public function setFitData(object $fit) {
```

Comment:
There is no 'general' class that all classes inherit from. As we know that the fit data should be an instance of `adriangibbons\phpFITFileAnalysis`, we can use that one as type hint.